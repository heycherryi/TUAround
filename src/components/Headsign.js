import React from 'react';
import PropTypes from 'prop-types';

function Headsign({ headsign }) {
  const matches = headsign.match(/^(\w{1,2})\s?-\s?(.*)/i);
  if (matches) {
    return (<span className="trip-headsign"><span className="badge text-bg-secondary">{matches[1]}</span> {matches[2]}</span>);
  }
  return (
    <span className="trip-headsign">{headsign}</span>
  );
}

Headsign.propTypes = {
  headsign: PropTypes.string.isRequired,
};

export default Headsign;
