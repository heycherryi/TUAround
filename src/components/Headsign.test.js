/* globals test, expect */

import React from 'react';
import { render, screen } from '@testing-library/react';
import Headsign from './Headsign';

test('renders Headsign', () => {
  const { container } = render(
    <Headsign headsign="DOWNTOWN" />,
  );
  expect(screen.getByText('DOWNTOWN')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('renders Headsign with Letter', () => {
  const { container } = render(
    <Headsign headsign="A - DOWNTOWN" />,
  );
  expect(screen.getByText('A')).toBeInTheDocument();
  expect(screen.getByText('DOWNTOWN')).toBeInTheDocument();
  expect(container.firstChild.firstChild.classList.contains('badge')).toBe(true);
  expect(container).toMatchSnapshot();
});

test('renders Headsign with Letter if missing trailing space', () => {
  const { container } = render(
    <Headsign headsign="A -WHITE BRIDGE" />,
  );
  expect(screen.getByText('A')).toBeInTheDocument();
  expect(screen.getByText('WHITE BRIDGE')).toBeInTheDocument();
  expect(container.firstChild.firstChild.classList.contains('badge')).toBe(true);
  expect(container).toMatchSnapshot();
});
