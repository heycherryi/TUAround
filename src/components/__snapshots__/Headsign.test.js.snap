// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Headsign 1`] = `
<div>
  <span
    class="trip-headsign"
  >
    DOWNTOWN
  </span>
</div>
`;

exports[`renders Headsign with Letter 1`] = `
<div>
  <span
    class="trip-headsign"
  >
    <span
      class="badge text-bg-secondary"
    >
      A
    </span>
     
    DOWNTOWN
  </span>
</div>
`;

exports[`renders Headsign with Letter if missing trailing space 1`] = `
<div>
  <span
    class="trip-headsign"
  >
    <span
      class="badge text-bg-secondary"
    >
      A
    </span>
     
    WHITE BRIDGE
  </span>
</div>
`;
